### ISSUE TITLE
Custom components configuration

### ISSUE THREAD

**Issue by susanhenriquezcp** [jsx-eslint/eslint-plugin-jsx-a11y#897]
Hello!
I am trying to make eslint-plugin-jsx-a11y in our project for accessibility. We are using Material UI and Next.js this is our linter configuration:
```
{
  "root": true,
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
    "next/core-web-vitals",
    "plugin:jsx-a11y/strict"
  ],
  "rules": {
    "@typescript-eslint/no-unused-vars": "error",
    "@typescript-eslint/no-explicit-any": "error",
    "@typescript-eslint/no-var-requires": "off"
  },
  "plugins": ["testing-library", "jsx-a11y"],
  "settings": {
    "jsx-a11y": {
      "components": {
        "Typography": {
          "props": {
            "variable": { "undefined": "p", "p": "p", "h1": "h1", "h2": "h2" }
          }
        },
        "Button": "button",
        "Switch": "input"
      }
    }
  }
}
```

But if I run the linter custom components does not seems to be affected. For example: I am having an empty <Typography></Typography> and it should get an error for being empty.

Thanks in advance!

**Comment by ljharb** [comment]
What version of the project are you using?

**Comment by rolule** [comment]
I just had the same issue because the plugin was installed as a dependency of another package. It worked once I installed it explicitly as a devDependency.

**Comment by susanhenriquezcp** [comment]
> What version of the project are you using?

@ljharb we are using "eslint-plugin-jsx-a11y": "6.6.1"

**Comment by susanhenriquezcp** [comment]
> I just had the same issue because the plugin was installed as a dependency of another package. It worked once I installed it explicitly as a devDependency.

We also having this as dev dependency but still not working. Are you also using Material UI? thanks

**Comment by ljharb** [comment]
I'm not using material UI, but eslint plugins must be peer dependencies everywhere that's not the top level.

**Comment by fwraymond** [comment]
Came to ask about this very issue. In digging into some of the rules it looks like some or all do not consider that settings key? For example looking at this function https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/4abc751d87a8491219a9a3d2dacd80ea8adcb79b/src/util/isInteractiveElement.js#L135 it only considers the elements coming from `dom` in `aria-query`

**Comment by ljharb** [comment]
Indeed, some of the rules haven't yet been updated to look at `settings`. A PR to fix that would be appreciated.
